src/main/java/org/spectrumauctions/sats/opt/model/gsvm/GSVMStandardMIP.java

```diff
@@ -40,11 +40,7 @@ public GSVMStandardMIP(List<GSVMBidder> population) {
 	}
 
 	public GSVMStandardMIP(GSVMWorld world, List<GSVMBidder> population) {
-		this(world, population, true);
-	}
-
-	public GSVMStandardMIP(GSVMWorld world, List<GSVMBidder> population, boolean allowAssigningLicensesWithZeroBasevalue) {
-		this.allowAssigningLicensesWithZeroBasevalue = allowAssigningLicensesWithZeroBasevalue;
+		this.allowAssigningLicensesWithZeroBasevalue = world.isLegacyGSVM();
```

Comment:
I changed the GSVMStandardMIP such that it responds to the worlds legacy setting. In this way it is possible to get the old behaviour by just changeing the world to legacy and efficient allocation, demand query and value query will work on the same value model.
Furthermore I added some tests for the demand query in legacy and non legacy setting.

Reply:
Great addition! Before, there was the possibility of an inconsistency if the user plays around with the parameters. Now, the single source of truth is the initially created world.